sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala

```diff
@@ -357,3 +360,35 @@ private[sql] object SortValue {
       None
   }
 }
+
+private[sql] sealed trait TimeTravelSpec
+
+private[sql] case class AsOfTimestamp(timestamp: Long) extends TimeTravelSpec
+private[sql] case class AsOfVersion(version: String) extends TimeTravelSpec
+
+private[sql] object TimeTravelSpec {
+  def create(timestamp: Option[String], version: Option[String]) : Option[TimeTravelSpec] = {
+    if (timestamp.nonEmpty && version.nonEmpty) {
+      throw new IllegalArgumentException(
+        "Version and Timestamp can't both be set in TimeTravelSpec")
+    } else if (timestamp.nonEmpty) {
+      val timeZoneId = DateTimeUtils.parseTimestampString(UTF8String.fromString(timestamp.get))._2
```

Comment:
We can simply do
```
val us = DateTimeUtils.stringToTimestampAnsi(
  UTF8String.fromString(timestamp.get),
  DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone))
```

If you look at the code, it respects the zone id in the input string.

Reply:
Changed. Thanks.